test/cql-pytest/test_tools.py

```diff
@@ -707,3 +708,142 @@ def test_external_dir_autodetect_conf_dir_home_env(self, scylla_path, system_scy
                 ext_sstable,
                 system_scylla_local_reference_dump,
                 env={"SCYLLA_HOME": scylla_home_dir})
+
+
+@pytest.fixture(scope="module")
+def scrub_schema_file():
+    """Create a schema.cql for the scrub tests"""
+    with tempfile.NamedTemporaryFile("w+t") as f:
```

Comment:
I'm not going to fight this now because I see there are alot of pre-existing examples of this "tempfile" stuff in this test file, so we can have a couple more, but I want to point out something:
The original philosophy of test/cql-pytest was to put all the temporary files stuff in one temporary directory (pid_to_dir() in run.py), which is automatically deleted when the test is interrupted. In the approach used here, the test creates a lot of temporary files, which we make an effort to remove (by correctly using "with") BUT we won't be able to remove if pytest itself is killed, and all this garbage will be left behind - which isn't terrible (it has unique names) but still ugly.
Note that in particular, interrupting test/cql-pytest/run today kills pytest with SIGKILL (`os.killpg(pid, 9)`) so will leave those temporary files behind. I don't know about test.py and how it kills pytest.

Reply:
I also support tests using the least amount of temporary directories and sharing these to the extent possible. I prefer separate temporary directory for each test, when the state of these directories are integral part of the tests, e.g. the test passes or fails based on how many or what files are in a directory. In this case, I want to avoid any mistakes causing a test to falsely pass/fail and so I resort to separate directory per test.

Reply:
A separate directory per test is fine, but it can be created as a subdirectory of the main Scylla temp directory instead of at the /tmp top level. But again, not important to change this now. Just a thought for the future.

Reply:
Indeed it is a good idea to have a common root. I will follow-up with a PR, fixing this and the other tests using separate temporary directories.